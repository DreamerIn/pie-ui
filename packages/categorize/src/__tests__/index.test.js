import Categorize from '../index';
import {
  ModelSetEvent,
  SessionChangedEvent
} from '@pie-framework/pie-player-events';

jest.mock('@pie-lib/math-rendering', () => ({ renderMath: jest.fn() }));

describe('categorize', () => {
  describe('events', () => {
    describe('model', () => {
      it('dispatches model set event', () => {
        const el = new Categorize();
        el.tagName = 'categorize-el';
        el.model = {};
        expect(el.dispatchEvent).toBeCalledWith(
          new ModelSetEvent('categorize-el', false, true)
        );
      });
    });
    describe('changeAnswers', () => {
      it('dispatches session changed event', () => {
        const el = new Categorize();
        el.tagName = 'categorize-el';
        el.session = { answers: [] };
        el.changeAnswers([{ foo: 'bar' }]);
        expect(el.dispatchEvent).toBeCalledWith(
          new SessionChangedEvent('categorize-el', true)
        );
      });
    });
  });
});
