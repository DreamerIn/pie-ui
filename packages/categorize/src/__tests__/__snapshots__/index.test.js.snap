// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`categorize renders snapshot 1`] = `
<ContextProvider
  value="903793"
>
  <div>
    <_default
      onToggle={[Function]}
      show={false}
      toggled={false}
    />
    <div
      style={
        Object {
          "flexDirection": "column-reverse",
        }
      }
    >
      <WithStyles(Categories)
        categories={Array []}
        grid={
          Object {
            "columns": NaN,
            "rows": NaN,
          }
        }
        model={
          Object {
            "categories": Array [],
            "choices": Array [],
            "correctResponse": Array [],
          }
        }
        onDropChoice={[Function]}
        onRemoveChoice={[Function]}
      />
      <WithStyles(Choices)
        choicePosition="above"
        choices={Array []}
        model={
          Object {
            "categories": Array [],
            "choices": Array [],
            "correctResponse": Array [],
          }
        }
      />
    </div>
  </div>
</ContextProvider>
`;

exports[`categorize renders snapshot with rationale 1`] = `
<ContextProvider
  value="899056"
>
  <div>
    <_default
      onToggle={[Function]}
      show={false}
      toggled={false}
    />
    <div
      style={
        Object {
          "flexDirection": "column-reverse",
        }
      }
    >
      <WithStyles(Categories)
        categories={Array []}
        grid={
          Object {
            "columns": NaN,
            "rows": NaN,
          }
        }
        model={
          Object {
            "categories": Array [],
            "choices": Array [],
            "correctResponse": Array [],
            "rationale": "This is rationale",
          }
        }
        onDropChoice={[Function]}
        onRemoveChoice={[Function]}
      />
      <WithStyles(Choices)
        choicePosition="above"
        choices={Array []}
        model={
          Object {
            "categories": Array [],
            "choices": Array [],
            "correctResponse": Array [],
            "rationale": "This is rationale",
          }
        }
      />
    </div>
    <Component
      labels={
        Object {
          "hidden": "Show Rationale",
          "visible": "Hide Rationale",
        }
      }
    >
      <div
        dangerouslySetInnerHTML={
          Object {
            "__html": "This is rationale",
          }
        }
      />
    </Component>
  </div>
</ContextProvider>
`;
